rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -488,25 +489,50 @@ private void saveNewUTXOs(BtcTransaction btcTx) throws IOException {
      */
     public void releaseBtc(Transaction rskTx) throws IOException {
 
-        //as we can't send btc from contracts we want to send them back to the sender
+        Coin value = rskTx.getValue().toBitcoin();
+        final RskAddress senderAddress = rskTx.getSender();
+        //as we can't send btc from contracts we want to send them back to the senderAddressStr
         if (BridgeUtils.isContractTx(rskTx)) {
             logger.trace("Contract {} tried to release funds. Release is just allowed from standard accounts.", rskTx);
-            throw new Program.OutOfGasException("Contract calling releaseBTC");
+            if (activations.isActive(ConsensusRule.RSKIP185)) {
+                refundAndEmitEvent(value, senderAddress, RejectedPegoutReason.CALLER_CONTRACT);
```

Comment:
should we refund a `sender`, if it is a contract? Looking at [No refund for contract calls](https://github.com/rsksmart/RSKIPs/blob/release-refund/IPs/RSKIP185.md#no-refund-for-contract-calls) section of the RSKIP..

Reply:
good question. @josedahlquist @marcos-iov what do you think?

Reply:
Good catch @Vovchyk! Yes, contracts calling the Bridge should not be refunded. Although the rejection event still needs to be emmited

Reply:
Ok, got it